### ISSUE TITLE
Godot Removes script_type from resource when loading on Windows for the first time

### ISSUE THREAD

**Issue by dparkinson101** [godotengine/godot#93192]
### Tested versions

- Reproducable in Godot Version 4.2.2

### System information

Godot v4.2.2.stable.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1070 Ti (NVIDIA; 31.0.15.5222) - Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz (12 Threads)

### Issue description

I have a resource which is a type of script object called a "RenderLayer" when opening on Mac this works fine and the resource is not modified, it looks like this:
```
[gd_resource type="Resource" script_class="RenderLayer" load_steps=13 format=3 uid="uid://csgcsd2sidnas"]
```gdscript

However when I load it for the first time on Windows after pulling it from a shared repo it converts to this:
```
[gd_resource type="Resource" load_steps=13 format=3 uid="uid://csgcsd2sidnas"]
```gdscript

This then causes problems as I have techinically modfied this item just by opening it and it loses the script reference if the .improt number changes causing and error on build. It also breaks Mac builds if it is accidentally commited.

### Steps to reproduce

On MacOS:
- Create a C# script defining a new type
- Create a new resource of that scriptable type
- Save project
On Windows:
- Load Project
- Select resource in Godot editor
- Check git changes and see resource has been modified as reported

### Minimal reproduction project (MRP)

Due to the nature of the bug creating a MRP project would not be applicable as I am not on MacOS and so cannot reproduce this on my own.

**Comment by KoBeWi** [comment]
Sounds like it's caused by missing .godot folder.

**Comment by dparkinson101** [comment]
So would we need to commit the something in the .godot folder to keep consistant resources between platforms? I'm a bit confused by this as isn't the .godot folder just created the first time a project is pulled from a repo loaded and all you should need to commit is the scripts and source / .import files for actual assets?

**Comment by dparkinson101** [comment]
Will try to get two projects from Mac and Windows respectively with the .godot folder for comparision in that case when I can, it might take a bit of time though as our team work in different time zones.

**Comment by KoBeWi** [comment]
No, .godot folder should not be committed. However problems are pretty much expected on first launch, as the cache data is being recreated. Be sure to restart the editor once it finishes, without saving anything. If this leads to any modification of the files then it's a bug.
